test/functional/tests/conftest.py

```diff
@@ -165,19 +167,23 @@ def base_prepare(item):
                 casadm.stop_all_caches()
                 casadm.remove_all_detached_cores()
             except Exception:
-                pass  # TODO: Reboot DUT if test is executed remotely
+                TestRun.executor.reboot()
+            if uninstall_cas:
+                installer.uninstall_opencas()
 
         for disk in TestRun.dut.disks:
             disk.umount_all_partitions()
             if not create_partition_table(disk, PartitionTable.gpt):
                 raise Exception(f"Failed to remove partitions from {disk}")
 
-        if get_force_param(item) and not TestRun.usr.already_updated:
+        if get_force_param(item) and not TestRun.usr.already_updated and not uninstall_cas:
             installer.rsync_opencas_sources()
             installer.reinstall_opencas()
-        elif not installer.check_if_installed():
+
+        elif not installer.check_if_installed() and not uninstall_cas:
```

Comment:
I would move `if` from :171 here and remove newly added conditions.
    `if uninstall_cas:`
         ` installer.uninstall_opencas()`
  `  elif get_force_param(item) and not TestRun.usr.already_updated:`
          `....`

please keep in mind `uninstall_opencas()` may raise an exception.